src/ai/backend/manager/registry.py

```diff
@@ -1582,6 +1586,54 @@ async def update_scaling_group(self, id, scaling_group) -> None:
         ) as rpc:
             await rpc.call.update_scaling_group(scaling_group)
 
+    async def calibrate_agent_alloc_differences(
+        self, kernel_agent_bindings: Sequence[KernelAgentBinding],
+    ):
+        """
+        Tries to calibrate agent row's occupied_slots with real value. This must be called
+        after kernel creation is completed, to prevent fraction of resource dropped by agent scheduler
+        during kernel creation still being reported as used.
+        """
+
+        def _str_to_decimal(orig_slots: ResourceSlot):
```

Comment:
Specify the return type.
How about renaming to `_numericalize`?
How about making it a method of `ResourceSlot`?

Reply:
Just found that `ResourceSlot.from_json()` converts value explicitly to `Decimal`. I'll remove this inner function and replace usages with `from_json()` calls.